### ISSUE TITLE
Using fixtures reusing aiohttp_client does not seem to work

### ISSUE THREAD

**Issue by Manfred1ni** [aio-libs/aiohttp#4387]
## Long story short

Hi everyone,

I'm trying to test my app with aiohttp, using the aiohttp_client. I managed to make it work for simple cases, but I run into an unexpected behaviour when using fixtures.
1. I have a the following fixture:

```
@pytest.fixture
def http_client(loop, app, aiohttp_client):
    return loop.run_until_complete(aiohttp_client(app))
```

2. When I use it in another fixture, I get the following error: `RuntimeError: Timeout context manager should be used inside a task`

It seems like the fixture is created outside a Task. But I don't understand why. The full trace is [here](https://gist.github.com/Manfred1ni/da15e2c1e9f7f307c09a47e1ad95c9e0).

However, it works when I __don't__ use a fixture event when I call `http_client` during the test.
## Expected behaviour

The fixture is created.

## Actual behaviour

The test setup fails.

## Steps to reproduce

Use the following gist: https://gist.github.com/Manfred1ni/da15e2c1e9f7f307c09a47e1ad95c9e0

## Your environment

- Python 3.6, on ubuntu 18.04
- the [gist](https://gist.github.com/Manfred1ni/da15e2c1e9f7f307c09a47e1ad95c9e0) also contains my `pip list`

**Comment by asvetlov** [comment]
`aiohttp_client` is **async** fixture, please use it inside an asynchronous context.
Something like
```
@pytest.fixture
async def http_client(loop, app, aiohttp_client):
    return await aiohttp_client(app)
```

**Comment by Manfred1ni** [comment]
Thank you for you reply @asvetlov.

As you suggested, I defined `http_client` as an async feature, removed the `loop.run_until_complete`, and marked my test as async too.

I still got the same error though, with the same trace. When looking into what's what, the `http_client` properly returns a TestClient, and a loop is attached to it.

Any other idea?

**Comment by asvetlov** [comment]
Enable strict warnings mode by `python -Werror` (or `filterwarnings = error` in `setup.cfg`).
It may help to detect *a lot* of problems.

Sorry, it is your code, not mine. I cannot access it, cannot run and figure out what's wrong.
A can say that aiohttp itself uses `aiohttp_client` fixture extensively.
On my job, we do the same a lot.
You are using the fixture a little different maybe if you've fallen into a problem.  
I have not enough information.
Basically, the error means that aiohttp client API was called from a sync function.

**Comment by Manfred1ni** [comment]
Finally had time to look a bit more into it. Thanks @asvetlov for your suggestion of using strict mode. That confirmed what you said: I was indeed doing stuff off-the-loop.

I have `pytest-aiohttp` and `pytest-asyncio` in my requirements. If I remove `pytest-asyncio`, the test passes. I'm still not quite sure why though. From pytest-asyncio's documentation, the library modifies the loop. But that does not explain why my fixture was used _outside_ a loop. Still, it's a start, I'll keep looking into it.
